app/assets/javascripts/controllers/ems_keypair/ems_keypair_controller.js

```diff
@@ -0,0 +1,63 @@
+ManageIQ.angular.app.controller('emsKeypairController', ['$http', '$scope', 'miqService', function($http, $scope, miqService) {
+  var init = function() {
```

Comment:
Can we please avoid using $scope in controllers? Check this article, it helps how to avoid using scope where it is not needed [Scope soup](http://www.technofattie.com/2014/03/21/five-guidelines-for-avoiding-scope-soup-in-angular.html).
Just replace `$scope.variable` with `this.variable` and then in  [_auth_keypair_angular_bootstrap.html.haml](https://github.com/mansam/manageiq/blob/replace-ems-sshkeypair-input-with-file-upload/app/views/layouts/angular-bootstrap/_auth_keypair_angular_bootstrap.html.haml)
define controller with as option `%div{"ng-controller" => "emsKeypairController as mv"}`, then you can use vm.variable in haml file.

The reason why this is better is that things in $scope makes it slower for angular to redraw and check for changes. Also you don't have problems with nested controllers with same variables.